lib/puppet/provider/package/yum.rb

```diff
@@ -23,6 +23,7 @@
       end
   end
 
+defaultfor :operatingsystem => :amazon
```

Comment:
@joshcooper would be ok to keep just this line for now and create another ticket to tackle the specificity issue?